Formula/manticoresearch.rb

```diff
@@ -18,21 +17,42 @@ class Manticoresearch < Formula
 
   depends_on "boost" => :build
   depends_on "cmake" => :build
-  depends_on "icu4c" => :build
-  depends_on "libpq" => :build
-  depends_on "mysql" => :build
+  depends_on "libpq"
+  depends_on "mysql"
   depends_on "openssl@1.1"
 
+  uses_from_macos "bison" => :build
+  uses_from_macos "flex" => :build
+
+  on_linux do
+    depends_on "gcc"
+  end
+
   conflicts_with "sphinx", because: "manticoresearch is a fork of sphinx"
 
+  fails_with gcc: "5"
+
   def install
     args = %W[
       -DCMAKE_INSTALL_LOCALSTATEDIR=#{var}
-      -DDISTR_BUILD=macosbrew
       -DBoost_NO_BOOST_CMAKE=ON
+      -DWITH_ICU=OFF
       -DWITH_ODBC=OFF
     ]
 
+    if OS.mac?
+      args << "-DDISTR_BUILD=macosbrew"
+    else
+      args += %W[
+        -DCMAKE_INSTALL_BINDIR=#{bin}
+        -DCMAKE_INSTALL_DATAROOTDIR=#{share}
+        -DCMAKE_INSTALL_INCLUDEDIR=#{include}
+        -DCMAKE_INSTALL_LIBDIR=#{lib}
+        -DCMAKE_INSTALL_MANDIR=#{man}
+        -DCMAKE_INSTALL_SYSCONFDIR=#{etc}
```

Comment:
Does the upstream not use defaults from `GNUInstallDirs`?

Reply:
I think it's because it assumes you are building a `.deb` file if you are on a Debian-based distro and uses the installation directories that these distros use.  It's unclear to me from the build instructions if there's a way to turn this off: https://github.com/manticoresoftware/manticoresearch/blob/fa3ecbd8adf9e2b1b781dd9b7ba56162ed426ff2/manual/Installation/Compiling_from_sources.md.